free-programming-books.md

```diff
@@ -496,6 +496,7 @@
 * [Small Memory Software](http://www.smallmemory.com/book.html)
 * [SVG Essentials](http://commons.oreilly.com/wiki/index.php/SVG_Essentials)
 * [The Quest for Artificial Intelligence: A History of Ideas and Achievements](http://ai.stanford.edu/~nilsson/QAI/qai.pdf) - Nils J. Nilsson (PDF)
+* [The Pragmatic Programmer, from Journeyman to Master](http://ricardogeek.com/docs/pragmatic_programmer.pdf) - Addison-Wesley 
```

Comment:
Where's the record of this being free to redistribute? The PDF linked has a copyright notice reserving all rights. There are banners throughout the PDF about ru.board which is a well-known "warez" site.